### PR TITLE
Babel plugin template app changes

### PR DESCRIPTION
**Pull request by tomduncalf**
## What, How & Why?

This PR contains the changes to the example and template apps to support the [Babel plugin](https://github.com/realm/realm-js/pull/4938).

These changes are split out because we cannot merge them right now, as we will need released versions of Realm and Realm React for the `package.json`, and possibly upgrading versions of React Native to match. Please ignore any changes to `package.json` in this PR, I will fix it up once we have a released version.

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: templates/expo-template-ts/app/models/Task.ts [hunk 1]
```diff
@@ -1,31 +1,18 @@
 import {Realm} from '@realm/react';
-export class Task extends Realm.Object {
-  _id!: Realm.BSON.ObjectId;
+
+// To use a class as a Realm object type in Typescript with the @realm/babel-plugin plugin,
```
Comment (kneth) [inline templates/expo-template-ts/app/models/Task.ts:3]:
```suggestion
// To use a class as a Realm object type in TypeScript with the @realm/babel-plugin plugin,
```

File: templates/react-native-template-realm-ts/template/app/models/Task.ts [hunk 2]
```diff
@@ -1,31 +1,18 @@
 import {Realm} from '@realm/react';
-export class Task extends Realm.Object {
-  _id!: Realm.BSON.ObjectId;
+
+// To use a class as a Realm object type in Typescript with the @realm/babel-plugin plugin,
```
Comment (kneth) [inline templates/react-native-template-realm-ts/template/app/models/Task.ts:3]:
```suggestion
// To use a class as a Realm object type in TypeScript with the @realm/babel-plugin plugin,
```

File: templates/scripts/sync_rn_ts_to_all_templates.sh [hunk 3]
```diff
@@ -9,16 +9,26 @@ set -o pipefail
...
   # to ".js" (regardless of whether the original was ".tsx" or ".ts")
   for i in `find . -path ./node_modules -prune -o -name "*.ts*" -print`; do npx detype $i $2/${i/\.ts*/\.js} ; done
+
   # Move the Babel config back
   mv babel.config.js.ignore babel.config.js
+
+  # Any files suffixed ".override.js" should override their automatically converted TS versions.
+  # This allows us to have a non-Babel plugin version of any models. *.override.js files are excluded
+  # from the final npm publish.
+  for i in `find . -path ./node_modules -prune -o -name "*.override.js" -print`; do mv $i $2/${i/\.override.js*/\.js} ; done
```
Comment (kneth) [inline templates/scripts/sync_rn_ts_to_all_templates.sh:30]: I consider `$(...)` better than  back ticks (`)

**Review by kneth — APPROVED** [review]

File: templates/scripts/sync_rn_ts_to_all_templates.sh (hunk 3 above)
Comment (tomduncalf) [reply templates/scripts/sync_rn_ts_to_all_templates.sh:30]: Ah I never even knew about `${...}`, thanks!

**Comment by tomduncalf** [comment]
Closing in favour of https://github.com/realm/realm-js/pull/5029/

**Pull request closed without merging**